src/Plugin/PasswordConstraint/MinimumCharacters.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Drupal\password_enhancements\Plugin\PasswordConstraint;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\PluginBase;
+use Drupal\password_enhancements\Plugin\PasswordConstraintPluginInterface;
+
+/**
+ * Defines a common class for minimum character requirement.
+ */
+abstract class MinimumCharacters extends PluginBase implements PasswordConstraintPluginInterface {
+
+  /**
+   * The generated error message.
+   *
+   * @var string
+   */
+  protected $errorMessage;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->errorMessage = NULL;
```

Comment:
I don't trust php ;) it doesn't hurt though.

Reply:
Then set NULL as the initial value where you define the property. In that case you do not need to override the constructor just to add `$this->errorMessage = NULL;`.

Reply:
If I'm right, an object should only declare its properties, but the initialization of those properties with default value (or computed values) should happen in the constructor.

Reply:
Scalar values can/should be initialized in the place of definition and I already suggested to you that do not the initialize `errorMessage` to `NULL` neither in the constructor, nor in the place of definition because it is already the default value of the property which is highly unlikely will change in the upcoming PHP versions .)

Reply:
Oh well, I cannot disprove this, going to remove the initialization.